packages/orchestration/src/exos/portfolio-holder-kit.js

```diff
@@ -0,0 +1,200 @@
+);
+
+/**
+ * A kit that holds several OrchestrationAccountKits and returns a invitation
+ * makers.
+ *
+ * @param {Zone} zone
+ * @param {VowTools} vowTools
+ */
+const preparePortfolioHolderKit = (zone, { watch }) => {
+  return zone.exoClassKit(
+    'PortfolioHolderKit',
+    {
+      invitationMakers: M.interface('InvitationMakers', {
+        Action: M.call(ChainNameM, M.string(), M.arrayOf(M.any())).returns(
+          VowShape,
+        ),
+      }),
+      holder: M.interface('Holder', {
+        asContinuingOffer: M.call().returns({
+          publicSubscribers: TopicsRecordShape,
+          invitationMakers: M.any(),
+        }),
+        getPublicTopics: M.call().returns(TopicsRecordShape),
+        getAccount: M.call(ChainNameM).returns(M.remotable()),
+        addAccount: M.call(
+          ChainNameM,
+          M.remotable(),
+          PublicTopicShape,
+        ).returns(),
+      }),
+      getInvitationMakersWatcher: M.interface('GetInvitationMakersWatcher', {
+        onFulfilled: M.call(
+          M.splitRecord({
+            invitationMakers: M.any(),
+            publicSubscribers: M.any(),
+          }),
+        )
+          .optional({
+            action: M.string(),
+            invitationArgs: M.arrayOf(M.any()),
+          })
+          .returns(M.promise()),
+      }),
+    },
+    /**
+     * @param {Iterable<[string, OrchestrationAccount<any>]>
+     *   | CopyMap<string, import('@endo/marshal').Passable>} accountEntires
+     * @param {Iterable<[string, ResolvedPublicTopic<unknown>]>
+     *   | CopyMap<string, import('@endo/marshal').Passable>} publicTopicEntires
+     */
+    (accountEntires, publicTopicEntires) => {
+      mustMatch(accountEntires, AccountEntriesShape, 'must provide accounts');
+      mustMatch(
+        publicTopicEntires,
+        PublicTopicsShape,
+        'must provide public topics',
+      );
+      const accounts = harden(
+        makeScalarBigMapStore('accounts', { durable: true }),
+      );
+      const publicTopics = harden(
+        makeScalarBigMapStore('publicTopics', { durable: true }),
+      );
+      accounts.addAll(accountEntires);
+      publicTopics.addAll(publicTopicEntires);
+      return /** @type {PortfolioHolderState} */ (
+        harden({
+          accounts,
+          publicTopics,
+        })
+      );
+    },
+    {
+      invitationMakers: {
+        /**
+         * @param {string} chainName key where the account is stored
+         * @param {string} action invitation maker name, e.g. 'Delegate'
+         * @param {unknown[]} invitationArgs
+         */
+        Action(chainName, action, invitationArgs) {
+          const { accounts } = this.state;
+          accounts.has(chainName) || Fail`no account found for ${chainName}`;
+          const account = accounts.get(chainName);
+          return watch(
+            E(account).asContinuingOffer(),
+            this.facets.getInvitationMakersWatcher,
+            { action, invitationArgs },
+          );
```

Comment:
`PortfolioHolderKit` is an endowment to the async-flow membrane, so want to make sure I've implemented that correctly.

1. Unlike [LOA](https://github.com/Agoric/agoric-sdk/blob/cd50120fb88b3179c53587d02512b04d2e146004/packages/orchestration/src/exos/local-orchestration-account.js#L296-L316) and [COA](https://github.com/Agoric/agoric-sdk/blob/cd50120fb88b3179c53587d02512b04d2e146004/packages/orchestration/src/exos/cosmos-orchestration-account.js#L315-L335), we don't need to make remote calls to retrieve public topics - they are already stored in state. This leads me to believe we can just return synchronously for `asContinuingOffer` and `getPublicTopics`. If this is not the case, we can wrap these in `asVow()`.

2. `Action`, which ultimately returns an invitation, currently returns a Vow. In LOA and COA, we return promises for invitations since the calls are expected to be prompt (and use vows for the offerHandler). Here, should we be returning a Promise since we expect the remote call to `.asContinuingOffer()` to resolve promptly?